### PR TITLE
Sm/potential-undefined

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
1. there's potentially `undefined` being hit here: https://github.com/salesforcecli/command/blob/84b3b5873eb7568bf05010303519a20c158fb90e/src/sfdxCommand.ts#L380 in the case where error is not an Error object, but a string.

Found via UT ex: https://github.com/salesforcecli/command/blob/84b3b5873eb7568bf05010303519a20c158fb90e/test/unit/sfdxCommand.test.ts#L841

2. maintenance dependency bump

### What issues does this PR fix or reference?
may help with some of the "ghost" undefineds happening in our error handling logic (ex? @W-11477173@)

definitely @W-11590389@ (linter WI)

### CODE REVIEW

File: .mocharc.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "require": "ts-node/register,source-map-support/register",
-    "watch-extensions": "ts",
-    "recursive": true,
-    "reporter": "spec",
-    "timeout": 5000
-  }
-  
\ No newline at end of file
+  "require": "ts-node/register,source-map-support/register",
```
Comment (WillieRuemmele) [inline .mocharc.json:2]: was this updated by `dev-scripts`?

File: src/docOpts.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
```
Comment (WillieRuemmele) [inline src/docOpts.ts:8]: is this rule enabled in the `tsconfig`? why do we have to add this to disable it in this file?

File: src/test/index.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
```
Comment (WillieRuemmele) [inline src/test/index.ts:7]: I don't understand what changed that would require this now

**Review by WillieRuemmele — APPROVED** [review]

File: src/docOpts.ts (hunk 2 above)
Comment (mshanemc) [reply src/docOpts.ts:8]: it's in the current eslint rules.  I struggled to understand some of this code and didn't want to change things I didn't understand.

File: .mocharc.json (hunk 1 above)
Comment (mshanemc) [reply .mocharc.json:2]: no, that's me.  I like to run `yarn mocha foo --watch`

If all our stuff was more standardized we probably could fix the `watch` property, but there's enough variations in test-naming (ex: sdr) and paths (toolbelt) etc.

**Comment by WillieRuemmele** [comment]
### QA Notes
---
by changing a command to `throw 'abc';`
now you get
`ERROR running force:data:soql:query:  abc`
before:
```
TypeError: Cannot read properties of undefined (reading 'replace')
    at DataSoqlQueryCommand.catch (/Users/william.ruemmele//sfdxCommand.js:264:66)
    at DataSoqlQueryCommand._run (/Users/william.ruemmele//sfdxCommand.js:93:29)
    at async Config.runCommand (/Users/william.ruemmele/@oclif/core/lib/config/config.js:272:25)
    at async Object.run (/Users/william.ruemmele/@oclif/core/lib/main.js:76:5)
```

**Pull request merged**